### PR TITLE
CC-493 addDisabilitiesRevised & ComboBox tests

### PR DESCRIPTION
**Pull request by tjheffner**
## Summary

adds basic unit test for combobox. difficult to test further in unit tests due to shadow dom from web components.

adds remaining tests for addDisabilitiesRevised page to bring coverage up.

## Related issue(s)

[https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/493](https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/493)

## Testing done

`yarn run test:coverage-app disability-benefits/all-claims` to see coverage report.

## Screenshots

before:
Screenshot 2024-05-30 at 11 21 31 AM
Screenshot 2024-05-30 at 11 22 02 AM

after:
Screenshot 2024-05-30 at 3 12 01 PM
Screenshot 2024-05-30 at 3 12 07 PM

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/tests/pages/addDisabilitiesRevised.unit.spec.jsx [hunk 1]
```diff
@@ -188,4 +190,66 @@ describe('showRevisedNewDisabilitiesPage', () => {
...
+      });
+    });
+
+    it('should not modify oldData', () => {
+      const old = oldData();
+      updateFormData(
+        old,
+        set('newDisabilities[1]', { condition: 'Something else' }, oldData()),
+      );
+      expect(old).to.deep.equal(oldData());
```
Comment (lukey-luke) [inline src/applications/disability-benefits/all-claims/tests/pages/addDisabilitiesRevised.unit.spec.jsx:231]: Can you help me understand what this is testing?

If we have old defined as 
```
old = {
  newDisabilities: [{ condition: 'Something with-hyphens and ALLCAPS' }],
  ...
};
```
Then call set() to update to
```
newObject = {
  newDisabilities: [
    { condition: 'Something with-hyphens and ALLCAPS' },
    { condition: 'Something else' },
  ],
  ...
}
```
why would `updateFormData(old, newObject)` return old?

File: src/applications/disability-benefits/all-claims/tests/components/ComboBox.unit.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  it('should render with a hidden list', () => {
+    const tree = render(<ComboBox {...defaultProps} />);
+    // nothing typed, no items visible
+    expect(tree.getByRole('listbox')).to.have.length(0);
+  });
+
+  // can't find the input element inside web-component shadowDOM
+  // in order to test any other functionality here...
+  // it('should handle keyboard input', async () => {
```
Comment (lukey-luke) [inline src/applications/disability-benefits/all-claims/tests/components/ComboBox.unit.spec.jsx:37]: makes sense 👍
Will try to connect synchronously to try to come up w/ a solution

File: src/applications/disability-benefits/all-claims/tests/pages/addDisabilitiesRevised.unit.spec.jsx (hunk 1 above)
Comment (tjheffner) [reply src/applications/disability-benefits/all-claims/tests/pages/addDisabilitiesRevised.unit.spec.jsx:231]: it's testing that that old data object is not mutated on an update

> why would updateFormData(old, newObject) return old?
it doesn't return old, but we are testing that there are no sideeffects to existing objects.

to be clear, these tests + functions are very similar to the ones on the existing addDisabilities page, but the tests hadn't been moved over.

Comment (lukey-luke) [reply src/applications/disability-benefits/all-claims/tests/pages/addDisabilitiesRevised.unit.spec.jsx:231]: I see what you're saying. Thanks for explaining.

**Comment by tjheffner** [comment]
> The linked ticket https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/493 includes updating the header to an <h3> is that change in a different PR or what ticket covers these changes?

there will be a follow-up PR will be coming shortly today for header changes. There wasn't a ticket for test coverage specifically, but it was raised in a few slack threads following the bug bash. 🙈

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Pull request merged**